docs/import/manage_disaster.js

```diff
@@ -369,71 +370,67 @@ function addDisaster() {
- *  write completes, their edit could be overwritten by the initial Firestore
- *  write here. Probably solved similar to the delete disaster issue: don't
- *  actually show the disaster as editable until this write completes.
  * @param {string} disasterId of the form <year>-<name>
  * @param {Array<string>} states array of state (abbreviations)
- * @return {Promise<boolean>} returns true after successful write to firestore
- * and earth engine folder creations.
+ * @return {Promise<boolean>} Returns true if EarthEngine folders created
+ *     successfully and Firestore write was successful
  */
-function writeNewDisaster(disasterId, states) {
+async function writeNewDisaster(disasterId, states) {
   if (disasterData.has(disasterId)) {
     setStatus('Error: disaster with that name and year already exists.');
-    return Promise.resolve(false);
+    return false;
   }
   clearStatus();
   const currentData = createDisasterData(states);
   disasterData.set(disasterId, currentData);
   // We know there are no assets in folder yet.
   disasterAssets.set(disasterId, Promise.resolve(new Map()));
 
-  const folderCreationPromises = [];
-  folderCreationPromises.push(
-      getCreateFolderPromise(eeLegacyPathPrefix + disasterId));
-  for (const state of states) {
-    folderCreationPromises.push(
-        getCreateFolderPromise(legacyStateDir + '/' + state));
+  const eeFolderPromises =
+      [getCreateFolderPromise(eeLegacyPathPrefix + disasterId)];
+  states.forEach(
+      (state) => eeFolderPromises.push(
+          getCreateFolderPromise(legacyStateDir + '/' + state)));
+
+  const tailError = '" You can try refreshing the page';
+  // Unsafe to do the Firestore write before EE folders are created, since maybe
+  // we won't be able to undo the write.
+  try {
+    await Promise.all(eeFolderPromises);
+  } catch (err) {
+    showError('Error creating EarthEngine folders: "' + err + tailError);
+    return false;
   }
-
-  const eePromisesResult = Promise.all(folderCreationPromises).then(() => {
-    const disasterPicker = $('#disaster-dropdown');
-    const disasterOptions = disasterPicker.children();
-    let added = false;
-    // We expect this recently created disaster to go near the top of the list,
-    // so do a linear scan down. Note: let's hope this tool isn't being used in
-    // the year 10000.
-    // Comment needed to quiet eslint.
-    disasterOptions.each(/* @this HTMLElement */ function() {
-      if ($(this).val() < disasterId) {
-        $(createOptionFrom(disasterId)).insertBefore($(this));
-        added = true;
-        return false;
-      }
-    });
-    if (!added) disasterPicker.append(createOptionFrom(disasterId));
-    toggleState(true);
-
-    disasterPicker.val(disasterId).trigger('change');
+  try {
+    await disasterCollectionReference().doc(disasterId).set(currentData);
+  } catch (err) {
+    const message = err.message ? err.message : err;
+    showError('Error writing to Firestore: "' + message + tailError);
```

Comment:
Should we do this here if we're not doing it everywhere?

Reply:
perfect enemy of the good? This seems better than the prior state, and I don't think it will share code with the other write, if we do add error handling there.